docs/proposals/20230407-flexible-managed-k8s-endpoints.md

```diff
@@ -0,0 +1,519 @@
+    - Must not wait for `spec.ControlPlaneEndpoint` field on the `Cluster` object to be set before starting to provision the control plane.
+    - As soon as the control plane endpoint is available, Must create a `ClusterEndpoint` CR to communicate the control plane endpoint to the Cluster API core controllers; the `ClusterEndpoint` CR must have an owner reference to the `FKRControlControlplane` object from which is originated.
+    - Must set the `status.Ready` field on the `FKRControlControlplane` object when the provisioning is complete
+
+_Scenario 3._
+
+If the `Foo` cloud provider has a `FKS` managed Kubernetes offering that is not taking care of the control plane endpoint e.g. because it requires an existing `FooElasticIP`, a `FooElacticLoadBalancer` to be provisioned before creating the `FKS` managed Kubernetes cluster, the maintainers of the `cluster-api-provider-foo` provider:
+- Must implement a `FKSCluster` CRD and the corresponding `FKSClusterTemplate` CRD and the related controllers; those controllers must create a `ClusterEndpoint` CR as soon as the control plane endpoint is available
+  - The `FKSCluster` controller
+    - Must create only the additional piece of the _Kubernetes Cluster infrastructure_ not provisioned by the `FKS` managed Kubernetes instance (in this example `FooElasticIP`, a `FooElacticLoadBalancer`)
+    - As soon as the control plane endpoint is available, Must create a `ClusterEndpoint` CR; the `ClusterEndpoint` CR must have an owner reference to the `FKSCluster` object from which is originated.
+    - Must set the `status.Ready` field on the `FKSCluster` object when the provisioning is complete
+- Must implement a `FKRControlControlplane provider`, a `FKRControlControlplane` CRD, the corresponding `FKRControlControlplane` and related controllers
+  - The `FKRControlControlplane` controller:
+    - Must wait for `status.InfrastructureReady` field on the `Cluster` object to be set to true before starting to provision the `FKS` managed Kubernetes instance.
+    - Must wait for `spec.ControlPlaneEndpoint` field on the `Cluster` object to be set before starting to provision the `FKS` managed Kubernetes instance.
+    - Must set the `status.Ready` field on the `FKRControlControlplane` object when the provisioning is complete
+
+Please note that this scenario is equivalent to what is implemented for a non managed Kubernetes `FooCluster`, backed by Cluster API managed `FooMachines`, with the only difference that in this case it possible to rely on `KCP` as `ControlControlplane provider`, and thus point 2 of the above list do not apply.
+
+## Background work
+
+This proposal builds on top of the awesome research work of the Managed Kubernetes working group, and it is a result of a huge work of a team of passionate Cluster API contributors.
+
+Below a summary of the main evidences / alternative considered during this this work.
```

Comment:
Do we need this considering the original proposal has been updated along with this PR?

Reply:
I agree with you that the information starting here is not helping clarify things (and possibly needs more cleaning up). I've asked @fabriziopandini if he thinks we can just remove it.

Reply:
Up to you folks, you did most of the background work, 
if you think that this doesn't help to clarify this proposal we can move it to a linked document (or let this implicit in the link to the updated original proposal)

Reply:
I'll remove for clarity. This was an attempt to restate the history that is already documented in other forms. In retrospect, asking someone to re-read this history again is too much.